test/e2e/policyrecommendation_test.go

```diff
@@ -0,0 +1,234 @@
+		testFlow := testFlow{
+			srcIP:      srcIP,
+			dstIP:      dstIP,
+			srcPodName: "perftest-a",
+			dstPodName: "perftest-b",
+		}
+		t.Run("testPolicyRecommendationResult/IPv6", func(t *testing.T) {
+			testPolicyRecommendationResult(t, data, true, testFlow)
+		})
+	}
+}
+
+// Example output: A new policy recommendation job is created successfully, id is e998433e-accb-4888-9fc8-06563f073e86
+func testPolicyRecommendationStart(t *testing.T, data *TestData) {
+	cmd, stdout, stderr := startJob(t, data)
+	assert := assert.New(t)
+	assert.Containsf(stdout, "A new policy recommendation job is created successfully, id is", "cmd:%s\nstdout:%s\nstderr:%s", cmd, stdout, stderr)
+}
+
+// Example output: Status of this policy recommendation job is COMPLETED
+func testPolicyRecommendationCheck(t *testing.T, data *TestData) {
+	_, stdout, _ := startJob(t, data)
+	stdoutSlice := strings.Split(stdout, " ")
+	jobId := strings.TrimSuffix(stdoutSlice[len(stdoutSlice)-1], "\n")
+	cmd, stdout, stderr := checkJob(t, data, jobId)
+	assert := assert.New(t)
+	assert.Containsf(stdout, "Status of this policy recommendation job is", "cmd:%s\nstdout:%s\nstderr:%s", cmd, stdout, stderr)
+}
+
+// Example output:
+// apiVersion: crd.antrea.io/v1alpha1
+// kind: NetworkPolicy
+// metadata:
+//   name: recommend-allow-anp-fj3hd
+// ...
+func testPolicyRecommendationResult(t *testing.T, data *TestData, isIPv6 bool, testFlow testFlow) {
+	var cmdStr string
+	if !isIPv6 {
+		cmdStr = fmt.Sprintf("iperf3 -c %s -t %d", testFlow.dstIP, iperfTimeSec)
+	} else {
+		cmdStr = fmt.Sprintf("iperf3 -6 -c %s -t %d", testFlow.dstIP, iperfTimeSec)
+	}
+	stdout, stderr, err := data.RunCommandFromPod(testNamespace, testFlow.srcPodName, "perftool", []string{"bash", "-c", cmdStr})
+	require.NoErrorf(t, err, "Error when running iPerf3 client: %v,\nstdout:%s\nstderr:%s", err, stdout, stderr)
+
+	_, stdout, _ = startJob(t, data)
+	stdoutSlice := strings.Split(stdout, " ")
+	jobId := strings.TrimSuffix(stdoutSlice[len(stdoutSlice)-1], "\n")
+	err = waitJobComplete(t, data, jobId, jobCompleteTimeout)
+	require.NoErrorf(t, err, "policy recommendation Spark job failed to completed")
+
+	// Apply the recommended policies, and check the results
+	getJobResult(t, data, jobId)
+	cmd := fmt.Sprintf("kubectl apply -f %s", outputYaml)
+	_, stdout, stderr, err = data.RunCommandOnNode(controlPlaneNodeName(), cmd)
+	require.NoErrorf(t, err, "Error when running %v from %s: %v\nstdout:%s\nstderr:%s", cmd, controlPlaneNodeName(), err, stdout, stderr)
+	cmd = fmt.Sprintf("kubectl get anp -n %s", testNamespace)
+	_, stdout, stderr, err = data.RunCommandOnNode(controlPlaneNodeName(), cmd)
+	require.NoErrorf(t, err, "Error when running %v from %s: %v\nstdout:%s\nstderr:%s", cmd, controlPlaneNodeName(), err, stdout, stderr)
+	outputLines := strings.Split(stdout, "\n")
+	allowCnt := 0
+	rejectCnt := 0
+	for _, line := range outputLines {
+		if strings.Contains(line, "recommend-allow") {
+			allowCnt += 1
```

Comment:
We'd better to seperate the count check for anp and acnp. For example:
allowAnpCnt, allowAcnpCnt, rejectAnpCnt, rejectAcnpCnt,

Reply:
thanks Yongming, updated